### ISSUE TITLE
GPU OOM in gigaspeech 

### ISSUE THREAD

**Issue by BuaaAlban** [espnet/espnet#3689]
**Describe the bug**
GPU OOM when training gigaspeech baseline using tesla V100 (32G)* 4 with conf 
"This configuration requires TITAN RTX (24GB) x 4 GPUs It takes about XX days"

**Basic environments:**
 - OS information: Ubuntu 18.04.1 LTS
 - python version: [e.g. 3.8.5 (default, Sep 24 2020, 16:55:52)  [GCC 7.5.0]`]
 - espnet version: [e.g. espnet 0.9.6]
 - Git hash [e.g. c84da5743b7ef70c0c6212715859bdebdcf873b2]
   - Commit date [e.g. Tue Sep 1 09:32:54 2020 -0400]
 - pytorch version [e.g. pytorch 1.7.1]
 

**Environments from `torch.utils.collect_env`:**

Collecting environment information...
PyTorch version: 1.7.1
Is debug build: False
CUDA used to build PyTorch: 10.2

OS: Ubuntu 18.04.1 LTS
GCC version: (GCC) 7.5.0
CMake version: version 3.10.2

Python version: 3.8
Is CUDA available: Yes
CUDA runtime version: 10.0.130
GPU models and configuration:
GPU 0: Tesla V100-PCIE
GPU 1: Tesla V100-PCIE
GPU 2: Tesla V100-PCIE
GPU 3: Tesla V100-PCIE

Nvidia driver version: 470.63.01
cuDNN version: /usr/lib/x86_64-linux-gnu/libcudnn.so.7.6.5

Versions of relevant libraries:
Versions of relevant libraries:
[pip3] numpy==1.20.1
[pip3] pytorch-wpe==0.0.1
[pip3] torch==1.7.1
[pip3] torch-complex==0.2.1
[conda] Could not collect

**Task information:**
 - Task: ASR
 - Recipe: gigaspeech
 - ESPnet2

**To Reproduce**
1. move to a recipe directory, e.g., `cd egs2/gigaspeech/asr1`
2. execute `run.sh` with specific arguments, e.g., `run.sh --stage 11 `
3. specify the error log, e.g., `exp/xxx/yyy.log`

**Config**
config: conf/train_asr.yaml
print_config: false
log_level: INFO
dry_run: false
iterator_type: sequence
output_dir: /group/dphi_algo_scratch_02/bokangz/exp/espnetgiga/exp/asr_train_asr_raw_en_char
ngpu: 1
seed: 0
num_workers: 1
num_att_plot: 3
dist_backend: nccl
dist_init_method: env://
dist_world_size: 4
dist_rank: 0
local_rank: 0
dist_master_addr: localhost
dist_master_port: 50089
dist_launcher: null
multiprocessing_distributed: true
cudnn_enabled: true
cudnn_benchmark: false
cudnn_deterministic: true
collect_stats: false
write_collected_feats: false
max_epoch: 20
patience: null
val_scheduler_criterion:
- valid
- loss
early_stopping_criterion:
- valid
- loss
- min
best_model_criterion:
-   - valid
    - acc
    - max
keep_nbest_models: 10
grad_clip: 5.0
grad_clip_type: 2.0
grad_noise: false
accum_grad: 4
no_forward_run: false
resume: true
train_dtype: float32
use_amp: false
log_interval: null
unused_parameters: false
use_tensorboard: true
use_wandb: false
wandb_project: null
wandb_id: null
pretrain_path: null
init_param: []
freeze_param: []
num_iters_per_epoch: null
batch_size: 20
valid_batch_size: null
batch_bins: 35000000
valid_batch_bins: null
train_shape_file:
- /group/dphi_algo_scratch_02/bokangz/exp/espnetgiga/exp/asr_stats_raw_en_char/train/speech_shape
- /group/dphi_algo_scratch_02/bokangz/exp/espnetgiga/exp/asr_stats_raw_en_char/train/text_shape.char
valid_shape_file:
- /group/dphi_algo_scratch_02/bokangz/exp/espnetgiga/exp/asr_stats_raw_en_char/valid/speech_shape
- /group/dphi_algo_scratch_02/bokangz/exp/espnetgiga/exp/asr_stats_raw_en_char/valid/text_shape.char
batch_type: numel
valid_batch_type: null
fold_length:
- 80000
- 150
sort_in_batch: descending
sort_batch: descending
multiple_iterator: false
chunk_length: 500
chunk_shift_ratio: 0.5
num_cache_chunks: 1024
train_data_path_and_name_and_type:
-   - /group/dphi_algo_scratch_02/bokangz/exp/espnetgiga/dump/raw/gigaspeech_train_xl/wav.scp
    - speech
    - kaldi_ark
-   - /group/dphi_algo_scratch_02/bokangz/exp/espnetgiga/dump/raw/gigaspeech_train_xl/text
    - text
    - text
valid_data_path_and_name_and_type:
-   - /group/dphi_algo_scratch_02/bokangz/exp/espnetgiga/dump/raw/gigaspeech_dev/wav.scp
    - speech
    - kaldi_ark
-   - /group/dphi_algo_scratch_02/bokangz/exp/espnetgiga/dump/raw/gigaspeech_dev/text
    - text
    - text
allow_variable_data_keys: false
max_cache_size: 0.0
max_cache_fd: 32
valid_max_cache_size: null
optim: adam
optim_conf:
    lr: 0.0015
scheduler: warmuplr
scheduler_conf:
    warmup_steps: 25000
token_list:
- <blank>
- <unk>
- <space>
- E
- T
- A
- O
- I
- N
- S
- H
- R
- L
- D
- U
- C
- M
- Y
- W
- G
- F
- P
- B
- V
- K
- ''''
- J
- X
- Q
- '-'
- Z
- <sos/eos>
init: null
input_size: null
ctc_conf:
    ignore_nan_grad: true
model_conf:
    ctc_weight: 0.3
    lsm_weight: 0.1
    length_normalized_loss: false
use_preprocessor: true
token_type: char
bpemodel: null
non_linguistic_symbols: null
cleaner: null
g2p: null
frontend: default
frontend_conf:
    n_fft: 512
    hop_length: 256
    fs: 16k
specaug: specaug
specaug_conf:
    apply_time_warp: true
    time_warp_window: 5
    time_warp_mode: bicubic
    apply_freq_mask: true
    freq_mask_width_range:
    - 0
    - 30
    num_freq_mask: 2
    apply_time_mask: true
    time_mask_width_range:
    - 0
    - 40
    num_time_mask: 2
normalize: global_mvn
normalize_conf:
    stats_file: /group/dphi_algo_scratch_02/bokangz/exp/espnetgiga/exp/asr_stats_raw_en_char/train/feats_stats.npz
preencoder: null
preencoder_conf: {}
encoder: conformer
encoder_conf:
    output_size: 512
    attention_heads: 8
    linear_units: 2048
    num_blocks: 12
    dropout_rate: 0.1
    positional_dropout_rate: 0.1
    attention_dropout_rate: 0.1
    input_layer: conv2d
    normalize_before: true
    macaron_style: true
    pos_enc_layer_type: rel_pos
    selfattention_layer_type: rel_selfattn
    activation_type: swish
    use_cnn_module: true
    cnn_module_kernel: 31
decoder: transformer
decoder_conf:
    attention_heads: 8
    linear_units: 2048
    num_blocks: 6
    dropout_rate: 0.1
    positional_dropout_rate: 0.1
    self_attention_dropout_rate: 0.1
    src_attention_dropout_rate: 0.1
required:
- output_dir
- token_list
distributed: true

**Error logs**
Paste the error logs. If applicable, add screenshots to help explain your problem
[train.log](https://github.com/espnet/espnet/files/7406899/train.log).
